quantum/rgblight.c

```diff
@@ -616,11 +624,79 @@ void rgblight_sethsv_master(uint8_t hue, uint8_t sat, uint8_t val) { rgblight_se
 void rgblight_sethsv_slave(uint8_t hue, uint8_t sat, uint8_t val) { rgblight_sethsv_range(hue, sat, val, (uint8_t)RGBLED_NUM / 2, (uint8_t)RGBLED_NUM); }
 #endif  // ifndef RGBLIGHT_SPLIT
 
+#ifdef RGBLIGHT_LAYERS
+static bool rgblight_is_static_mode(void) {
+    uint8_t mode = rgblight_status.base_mode;
+    bool is_static = 
+#   ifdef RGBLIGHT_EFFECT_STATIC_GRADIENT
+                    mode == RGBLIGHT_MODE_STATIC_GRADIENT ||
+#   endif
+                    mode == RGBLIGHT_MODE_STATIC_LIGHT;
+    return is_static;
+}
+
+void rgblight_set_layer_state(uint8_t layer, bool enabled) {
+    uint8_t mask = 1 << layer;
+    if (enabled) {
+        rgblight_status.enabled_layer_mask |= mask;
+    } else {
+        rgblight_status.enabled_layer_mask &= ~mask;
+    }
+#    ifdef RGBLIGHT_SPLIT
+    rgblight_status.change_flags |= RGBLIGHT_STATUS_CHANGE_LAYERS;
+#    endif
+    // Static modes don't have a ticker running to update the LEDs
+    if (rgblight_is_static_mode()) {
```

Comment:
Use is_static_effect () as it already exists.

```suggestion
    if (is_static_effect(rgblight_config.mode)) {
```

Reply:
Thanks, I found an even better way: just look at `rgblight_status.timer_enabled`, since that's what I actually care about.